### ISSUE TITLE
grid sample problem 

### ISSUE THREAD

**Issue by sudospike** [Tencent/TNN#957]
on ubuntu 18.04 cpu
pytorch->onnx->tnn grid_sample problem

export the model with the grid sample operation and convert onnx to tnn
it failed when inference with tnnproto and tnnmodel

what is the right way to do with grid_sample

error log:

E/tnn: Init [File /home/yyc/frame/tnn/TNN/source/tnn/layer/base_layer.cc][Line 82] layer acc of type(323) is nil
E/tnn: InitLayers [File /home/yyc/frame/tnn/TNN/source/tnn/core/default_network.cc][Line 319] Error Init layer output (err: 12288 or 0x3000)
E/tnn: Init [File /home/yyc/frame/tnn/TNN/source/tnn/layer/base_layer.cc][Line 82] layer acc of type(323) is nil
E/tnn: InitLayers [File /home/yyc/frame/tnn/TNN/source/tnn/core/default_network.cc][Line 319] Error Init layer output (err: 12288 or 0x3000)
TNN API ERROR:0x3000E/tnn: Init [File /home/yyc/frame/tnn/TNN/source/tnn/layer/base_layer.cc][Line 82] layer acc of type(323) is nil
E/tnn: InitLayers [File /home/yyc/frame/tnn/TNN/source/tnn/core/default_network.cc][Line 319] Error Init layer output (err: 12288 or 0x3000)

**Comment by sudospike** [comment]
[grid_sample.zip](https://github.com/Tencent/TNN/files/6486022/grid_sample.zip)

**Comment by sudospike** [comment]
Can anybody help me with the example of the mini model?

**Comment by sudospike** [comment]
i get onnx use code like this `import torch
from torch.onnx import OperatorExportTypes
import torch.nn.functional as F
import torch.nn as nn

def conv(in_planes, out_planes, kernel_size=3, stride=1, padding=1, dilation=1):
    return nn.Conv2d(in_planes, out_planes, kernel_size=kernel_size, stride=stride,
                  padding=padding, dilation=dilation, bias=True)


class MyModel(torch.nn.Module):
    
    def __init__(self):
        super(MyModel, self).__init__()
        self.flownet = conv(3, 2, 3, 1, 1)

    def forward(self, x):
        flow = self.flownet(x)
        flow = flow.permute(0, 2, 3, 1)
        return F.grid_sample(x, flow)

x = torch.rand(1, 3, 10, 10)
model = MyModel()

torch.onnx.export(
    model,
    (x),
    "grid_sample.onnx",
    verbose=True,
    input_names=['frame'],
    output_names=['warped'],
    operator_export_type=OperatorExportTypes.ONNX_ATEN_FALLBACK
)


`

**Comment by sudospike** [comment]
`import torch
from torch.onnx import OperatorExportTypes
import torch.nn.functional as F
import torch.nn as nn

def conv(in_planes, out_planes, kernel_size=3, stride=1, padding=1, dilation=1):
    return nn.Conv2d(in_planes, out_planes, kernel_size=kernel_size, stride=stride,
                  padding=padding, dilation=dilation, bias=True)


class MyModel(torch.nn.Module):
    
    def __init__(self):
        super(MyModel, self).__init__()
        self.flownet = conv(3, 2, 3, 1, 1)

    def forward(self, x):
        flow = self.flownet(x)
        flow = flow.permute(0, 2, 3, 1)
        return F.grid_sample(x, flow)

x = torch.rand(1, 3, 10, 10)
model = MyModel()

torch.onnx.export(
    model,
    (x),
    "grid_sample.onnx",
    verbose=True,
    input_names=['frame'],
    output_names=['warped'],
    operator_export_type=OperatorExportTypes.ONNX_ATEN_FALLBACK
)


`

**Comment by sudospike** [comment]
`
import torch
from torch.onnx import OperatorExportTypes
import torch.nn.functional as F
import torch.nn as nn

def conv(in_planes, out_planes, kernel_size=3, stride=1, padding=1, dilation=1):
    return nn.Conv2d(in_planes, out_planes, kernel_size=kernel_size, stride=stride,
                  padding=padding, dilation=dilation, bias=True)


class MyModel(torch.nn.Module):
    
    def __init__(self):
        super(MyModel, self).__init__()
        self.flownet = conv(3, 2, 3, 1, 1)

    def forward(self, x):
        flow = self.flownet(x)
        flow = flow.permute(0, 2, 3, 1)
        return F.grid_sample(x, flow)

x = torch.rand(1, 3, 10, 10)
model = MyModel()

torch.onnx.export(
    model,
    (x),
    "grid_sample.onnx",
    verbose=True,
    input_names=['frame'],
    output_names=['warped'],
    operator_export_type=OperatorExportTypes.ONNX_ATEN_FALLBACK
)

`

**Comment by sudospike** [comment]
import torch
from torch.onnx import OperatorExportTypes
import torch.nn.functional as F
import torch.nn as nn

def conv(in_planes, out_planes, kernel_size=3, stride=1, padding=1, dilation=1):
    return nn.Conv2d(in_planes, out_planes, kernel_size=kernel_size, stride=stride,
                  padding=padding, dilation=dilation, bias=True)


class MyModel(torch.nn.Module):
    
    def __init__(self):
        super(MyModel, self).__init__()
        self.flownet = conv(3, 2, 3, 1, 1)

    def forward(self, x):
        flow = self.flownet(x)
        flow = flow.permute(0, 2, 3, 1)
        return F.grid_sample(x, flow)

x = torch.rand(1, 3, 10, 10)
model = MyModel()

torch.onnx.export(
    model,
    (x),
    "grid_sample.onnx",
    verbose=True,
    input_names=['frame'],
    output_names=['warped'],
    operator_export_type=OperatorExportTypes.ONNX_ATEN_FALLBACK
)

**Comment by sudospike** [comment]
import torch
  from torch.onnx import OperatorExportTypes
  import torch.nn.functional as F
  import torch.nn as nn
  
  def conv(in_planes, out_planes, kernel_size=3, stride=1, padding=1, dilation=1):
      return nn.Conv2d(in_planes, out_planes, kernel_size=kernel_size, stride=stride,
                    padding=padding, dilation=dilation, bias=True)
  
  
  class MyModel(torch.nn.Module):
      
      def __init__(self):
          super(MyModel, self).__init__()
          self.flownet = conv(3, 2, 3, 1, 1)
  
      def forward(self, x):
          flow = self.flownet(x)
          flow = flow.permute(0, 2, 3, 1)
          return F.grid_sample(x, flow)
  
  x = torch.rand(1, 3, 10, 10)
  model = MyModel()
  
  torch.onnx.export(
      model,
      (x),
      "grid_sample.onnx",
      verbose=True,
      input_names=['frame'],
      output_names=['warped'],
      operator_export_type=OperatorExportTypes.ONNX_ATEN_FALLBACK
  )

**Comment by sudospike** [comment]
`import torch
from torch.onnx import OperatorExportTypes
import torch.nn.functional as F
import torch.nn as nn

def conv(in_planes, out_planes, kernel_size=3, stride=1, padding=1, dilation=1):
    return nn.Conv2d(in_planes, out_planes, kernel_size=kernel_size, stride=stride,
                  padding=padding, dilation=dilation, bias=True)


class MyModel(torch.nn.Module):
    
    def __init__(self):
        super(MyModel, self).__init__()
        self.flownet = conv(3, 2, 3, 1, 1)

    def forward(self, x):
        flow = self.flownet(x)
        flow = flow.permute(0, 2, 3, 1)
        return F.grid_sample(x, flow)

x = torch.rand(1, 3, 10, 10)
model = MyModel()

torch.onnx.export(
    model,
    (x),
    "grid_sample.onnx",
    verbose=True,
    input_names=['frame'],
    output_names=['warped'],
    operator_export_type=OperatorExportTypes.ONNX_ATEN_FALLBACK
)


`

**Comment by sudospike** [comment]
i use the code to generate onnx
```
import torch
from torch.onnx import OperatorExportTypes
import torch.nn.functional as F
import torch.nn as nn

def conv(in_planes, out_planes, kernel_size=3, stride=1, padding=1, dilation=1):
    return nn.Conv2d(in_planes, out_planes, kernel_size=kernel_size, stride=stride,
                  padding=padding, dilation=dilation, bias=True)


class MyModel(torch.nn.Module):
    
    def __init__(self):
        super(MyModel, self).__init__()
        self.flownet = conv(3, 2, 3, 1, 1)

    def forward(self, x):
        flow = self.flownet(x)
        flow = flow.permute(0, 2, 3, 1)
        return F.grid_sample(x, flow)

x = torch.rand(1, 3, 10, 10)
model = MyModel()

torch.onnx.export(
    model,
    (x),
    "grid_sample.onnx",
    verbose=True,
    input_names=['frame'],
    output_names=['warped'],
    operator_export_type=OperatorExportTypes.ONNX_ATEN_FALLBACK
)
```

**Comment by gttiankai** [comment]
What is your target platform? try arm devices。
this PR（https://github.com/Tencent/TNN/pull/931） support grid_sample for arm devices.

**Comment by lnmdlong** [comment]
https://github.com/Tencent/TNN/pull/1095 support grid_sample for opencl
https://github.com/Tencent/TNN/pull/1189 support grid sample for metal
